tests/test_config_parser.py

```diff
@@ -58,3 +59,38 @@ class Foo:
     assert "initial comment" in spec.initial_comment[0]
     assert "final comment" in spec.final_comment[0]
     assert "inline comment (with parentheses)" in spec.inline_comments["bar"]
+
+
+@pytest.mark.parametrize(
+    "action", ["default", "error", "ignore", "always", "module", "once"]
+)
+def test_validate_extra_value_warning(action):
+    """Test that extra values in the configuration raise warnings.
+
+    The warning behavior can be configured by modifying the
+    EXTRA_VALUE_WARNING_ACTION switch for the module
+    """
+    config = ConfigObj({"expected": True, "unexpected": False})
+
+    class MockStep:
+        spec = "expected = boolean(default=False) # Expected parameter"
+
+    spec = config_parser.load_spec_file(MockStep)
+    config_parser.EXTRA_VALUE_WARNING_ACTION = action
```

Comment:
A [monkeypatch](https://docs.pytest.org/en/6.2.x/monkeypatch.html) would be useful here to reset `EXTRA_VALUE_WARNING_ACTION` back to the default value when the test finishes.

Reply:
Good point!  I'll fix that.